### ISSUE TITLE
v4 macro-epic

### ISSUE THREAD

**Issue by trentmc** [oceanprotocol/multi-repo-issue#93]
- [ ] Move all unit tests in `contracts` to `ocean.js` repo, and reconcile. DRY. [Andrea]
- [ ] Finish ocean.py pre-v4 refactorings [Trent, Maria]
- [ ]

**Comment by jamiehewitt15** [comment]
Is the stable coin task part of V4?

**Comment by trentmc** [comment]
No, it's part of the cluster of work "frontend flexibility". (v4 = "backend flexibility)

I will DM you a link to the slides with more info
